src/vivarium_csu_zenon/components/disease.py

```diff
@@ -114,18 +121,129 @@ def DiabetesMellitus():
+            disability_weight_data = builder.data.load(disability_weight_key)
+            self.base_disability_weight[ikf_category] = builder.lookup.build_table(disability_weight_data,
+                                                                                   key_columns=['sex'],
+                                                                                   parameter_columns=['age', 'year'])
+            self.disability_weight[ikf_category] = builder.value.register_value_producer(
+                f'{project_globals.IKF_TO_CKD_MAP[ikf_category]}.disability_weight',
+                source=lambda index: self.compute_disability_weight(index, ikf_category),
+                requires_values=[f'{project_globals.IKF.name}.exposure']
+            )
+
+            builder.value.register_value_modifier('disability_weight', modifier=self.disability_weight[ikf_category])
+
+        excess_mortality_data = builder.data.load(project_globals.IKF.EMR)
+        self.base_excess_mortality_rate = {}
+        self.excess_mortality_rate = {}
+        for ikf_category, ckd_state in project_globals.IKF_TO_CKD_MAP.items():
+            # Calculate EMR
+            self.base_excess_mortality_rate[ikf_category] = builder.lookup.build_table(
+                excess_mortality_data, key_columns=['sex'], parameter_columns=['age', 'year'])
+
+            self.excess_mortality_rate[ikf_category] = builder.value.register_rate_producer(
+                f'{ckd_state}.excess_mortality_rate',
+                source=lambda index: self.compute_excess_mortality_rate(index, ikf_category),
+                requires_values=[f'{project_globals.IKF.name}.exposure']
+            )
+
+            # Calculate mortality rate adjustment
+            builder.value.register_value_modifier(
+                'mortality_rate',
+                modifier=lambda index, rates: self.adjust_mortality_rate(index, rates, ikf_category),
+                requires_values=[f'{ckd_state}.excess_mortality_rate']
+            )
+
+        self.ikf_exposure = builder.value.get_value(f'{project_globals.IKF.name}.exposure')
+
+    def on_initialize_simulants(self, pop_data: 'SimulantData'):
+        exposure = self.ikf_exposure(pop_data.index)
+        states = exposure.map(project_globals.IKF_TO_CKD_MAP)
+        states.name = self.name
+        self.population_view.update(states)
+
+    def adjust_cause_specific_mortality_rate(self, index, rate):
+        return rate + self.cause_specific_mortality_rate(index)
+
+    # leave this here
+    def load_disability_weight_data(self, builder: 'Builder'):
+        dfs = []
+        for i, disability_weight_key in enumerate(project_globals.IKF.disability_weights):
+            col_name = f'cat{i+1}'
+            df = builder.data.load(disability_weight_key)
+            df = df.rename(columns={'value': col_name})
+            df = df.set_index([c for c in df.columns if c != col_name])
+            dfs.append(df)
+
+        cat5 = dfs[-1].copy()
```

Comment:
I'd suggest doing this over on the artifact building side.

Reply:
that way the data that the researchers pull out is reflective of what's in the model.

Reply:
Nevermind.